src/index.ts

```diff
@@ -0,0 +1,206 @@
+      const itemId = id;
+      // get actions aplying the parameters (view and requestedSampleSize)
+      const t1 = new GetActionsTask(member, itemId, requestedSampleSize, view, actionService);
+      await runner.runSingle(t1);
+      const actions = t1.result;
+      console.log(actions)
+
+      // get item
+      const t2 = itemTaskManager.createGetTaskSequence(member, itemId);
+      const itemResponse = await runner.runSingleSequence(t2);
+      const item = itemResponse as Item;
+      console.log(item)
+
+      // get memberships of the item
+      const t3 = itemMembershipsTaskManager.createGetOfItemTaskSequence(member, itemId);
+      const membershipsResponse = await runner.runSingleSequence(t3);
+      const memberships = membershipsResponse as ItemMembership[];
+      console.log(memberships)
+
+      // get members of the item
+      const tasks = memberships.map((membership) => memberTaskManager.createGetTask(member, membership.memberId));
+      const membersResponse = await runner.runMultiple(tasks, log);
+      const members = membersResponse as Member[];
+      console.log(members)
+
+      const numActionsRetrieved = actions.length;
+      const metadata = {
+        numActionsRetrieved: numActionsRetrieved,
+        requestedSampleSize: parseInt(requestedSampleSize)
+      }
+
+      // generate responseData with actions, members, item, and metadata
+      const responseData: Analytics = new BaseAnalytics(actions, members, item, metadata);
+      
+      reply
+      .send( responseData )
+    },
+  );
+  
+  // delete all the actions matching the given `memberId`
+  fastify.delete<{ Params: IdParam }>(
+    '/members/:id/delete-actions',
+    { schema: getOneDeleteActions },
+    async ({ member, params: { id }, log,  }, reply) => {
+      const task = new DeleteActionsTask(member, id, actionService);
+      await runner.runSingle(task);
+
+      reply
+      .send( task.result )      
+    },
+  );
+
+};
+
+
+// save action (except create and delete) for the itemId with its actionType and view
+const createActionTask = async function (member: Member, itemId: string, actionType: string, view: string, geo, runner, itemTaskManager) {
+  // get item
+  const t1 = itemTaskManager.createGetTaskSequence(member, itemId);
+  const item = await runner.runSingleSequence(t1);
+  // save action
+  const extra = {memberId: member.id, itemId: item.id};
+  let action: Action = new BaseAction(member.id, item.id, member.type, item.type, actionType, view, geo, extra);
+  const t2 = new CreateActionTask(member, action, actionService);
+  await runner.runSingle(t2);
+};
+
+// function called each time there is a request in the items in graasp (onResponse hook in graasp)
+// identify and check the correct endpoint of the request
+export const createAction = async function (request, reply, runner, itemTaskManager) {
+  // check that the request is ok
```

Comment:
If these don't depend on the plugin, maybe it would be nice to factor them out in another file.